### PR TITLE
Documentation cleanup/renaming effort

### PR DESCRIPTION
**Pull request by maxinelasp**
# Change Summary
Just wanted to clean up and rearrange some docs, and add a link to the new NASA TOPS open science 101 course. I'd appreciate any notes on naming changes or locations.

## Overview

Renamed a few top level files to be more specific, and moved files that I thought made more sense in different areas. With :ref: this didn't make much difference in the actual contents of the files, although I did tidy up a few index.rst files. 

## Text changes

- top level index.rst file
- code-documentation/tools/cdf-global-attrs
   - Just added a few sentences here, mostly wanted to add it so I can start fleshing this out in a later PR

### CODE REVIEW

**Comment by maxinelasp** [comment]
It can be hard to figure out the changes in the github view so here's the renaming:

Big name changes:
`reference` is renamed to `code-documentation`
`development` is renamed to `development-guide`
Created `external-tools`

Moves:
moved CDF and poetry docs to `external-tools`
Moved xarray to cdf and xtce documentation into `code-documentation/tools`

So, the main top-level directories are now:
`code-documentation`
`development-guide`
`external-tools`
`project-mangement` (unchanged in this PR)

Which I feel makes it more clear to external users what the documentation groups contain and make it easier to dig down and find what you're looking for.

File: .gitignore [hunk 1]
```diff
@@ -1,5 +1,5 @@
 # Documentation ignore
-docs/source/reference/generated
+docs/source/instrument-documentation/generated
```
Comment (bourque) [inline .gitignore]: Should this be `code-documentation`?

**Review by tech3371 — APPROVED** [review]
I clicked through the document page and makes sense to me

**Review by bourque — APPROVED** [review]
Nice, I think these are an improvement for sure. Thanks for doing that!

File: .gitignore (hunk 1 above)
Comment (maxinelasp) [reply .gitignore]: Yup, good catch

Comment (maxinelasp) [reply .gitignore]: Odd that the intellij refactor renamed it the first time but not the second time 🤔

**Comment by bourque** [comment]
@all-contributors please add @maxinelasp for maintenance

**Pull request merged**